agenta-backend/agenta_backend/tests/unit/test_evaluators.py

```diff
@@ -335,3 +340,133 @@ def test_auto_levenshtein_distance(output, data_point, settings_values, expected
+                },
+                "outputs": {
+                    "report": "These three films explore the complex relationship between humans and artificial intelligence. In 'Ex Machina,' a programmer interacts with a humanoid AI, questioning consciousness and morality. 'I Am Mother' features a girl raised by a robot in a post-extinction world, who challenges her understanding of trust and the outside world when a human arrives. 'Mother/Android' follows a pregnant woman and her boyfriend navigating a post-apocalyptic landscape controlled by hostile androids, highlighting themes of survival and human resilience."
+                },
+                "locals": {},
+            },
+        ],
+    },
+}
+
+
+@pytest.mark.parametrize(
+    "evaluator_function, expected_name, settings_values, expected_score",
+    [
+        (
+            rag_faithfulness,
+            "Faithfulness",
+            {
+                "question_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.retriever.outputs.prompt",
+                },
+                "answer_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.generator.outputs.report",
+                },
+                "contexts_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.retriever.outputs.movies",
+                },
+            },
+            0.0,
+        ),
+        (
+            rag_context_relevancy,
+            "Context Relevancy",
+            {
+                "question_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.retriever.outputs.prompt",
+                },
+                "answer_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.generator.outputs.report",
+                },
+                "contexts_key": {
+                    # ...
+                    "type": "string",
+                    "default": "rag.retriever.outputs.movies",
+                },
+            },
+            0.9,
+        ),
+    ],
+)
+def test_rag_evaluator(
+    evaluator_function, expected_name, settings_values, expected_score
+):
+    question_key = get_user_key_from_settings(settings_values, "question_key")
+    answer_key = get_user_key_from_settings(settings_values, "answer_key")
+    contexts_key = get_user_key_from_settings(settings_values, "contexts_key")
+
+    question_value = get_field_value_from_trace(app_output, question_key)
+    answer_value = get_field_value_from_trace(app_output, answer_key)
+    contexts_value = get_field_value_from_trace(app_output, contexts_key)
+
+    result = evaluator_function(
+        {},
+        app_output,
+        {},
+        {},
+        settings_values,
+        {},
+    )
+
+    assert round(result.value, 1) == expected_score
```

Comment:
I've seen it go from 0.3333 to 1 depending on the LLM hallucination. With a strict 1, this is a flaky test. Lets simply test that it return a score that is between 0 and 1. No ?

Reply:
Mokes sense.